### DISCUSSION TITLE
Using workers.extraContainers with KubernetesExecutor occasional termination failures

### DISCUSSION THREAD

**Discussion by sbakiu** [apache/airflow#23978] [Q&A]
I am using a sidecar container for my tasks configured via the `workers.extraContainers` section in the helm chart. 
In the normal flow, I have overridden the Operator's `execute` function to create a flag file that triggers the termination of the sidecar container. 

However, time-to-time it happens that the operator does not start/finish normally, usually with an `AirflowException` or similar, and the `base` container is terminated while the `sidecar` container is still running. This blocks Airflow from marking the task as `up_for_retry`. What would be the recommended way to make sure that the flag file is created, no matter if the task succeeded or failed?
Would you consider adding some more pointers in the documentation as well (https://airflow.apache.org/docs/helm-chart/stable/using-additional-containers.html#sidecar-containers)?

Thanks in advance

**Comment by potiuk** [comment]
You can use `on_kill()` - and yes, feel free to add it to the docs yourself. Airflow (including the docs) is created by more than 2000 contributors and you can easily become one of them if you contribute that documentation piece.

It is super easy - you already know where to add it. Go to the page and click "suggest a change on this page" and you will get PR opened, where you will be able to add documentation directly in GitHub UI. It's as easy as opening a discussion or issue in GitHub.
